pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -220,18 +239,56 @@ func injectLibRequirements(pod *corev1.Pod, ctrName string, envKey string, envVa
 			continue
 		}
 
-		index := envIndex(ctr.Env, envKey)
-		if index < 0 {
-			pod.Spec.Containers[i].Env = append(pod.Spec.Containers[i].Env, corev1.EnvVar{
-				Name:  envKey,
-				Value: envVal(""),
-			})
-		} else {
-			if pod.Spec.Containers[i].Env[index].ValueFrom != nil {
-				return fmt.Errorf("%q is defined via ValueFrom", envKey)
+		err := setEnvVarWithFunc(&pod.Spec.Containers[i], envKey, envVal)
+		if err != nil {
+			return err
+		}
+
+		volumeAlreadyMounted := false
+		for _, vol := range pod.Spec.Containers[i].VolumeMounts {
+			if vol.Name == volumeName {
+				volumeAlreadyMounted = true
+				break
 			}
+		}
+		if !volumeAlreadyMounted {
+			pod.Spec.Containers[i].VolumeMounts = append(pod.Spec.Containers[i].VolumeMounts, corev1.VolumeMount{Name: volumeName, MountPath: mountPath})
+		}
+	}
+
+	return nil
+}
+
+// injectLibRequirements injects the minimal config requirements to enable instrumentation
+func injectLibRequirementsDotnet(pod *corev1.Pod, ctrName string) error {
```

Comment:
It seems like something like:

```
type envVar struct {
  key string
  valFunc envValFunc
}

func injectLibRequirements(pod *corev1.Pod, ctrName string, envVars []enVar) error
```

Would be more suited

Reply:
I could implement this, but I would also require creating a total of 5 new functions to set the .NET environment variables. Is that a trade-off you think is worth making?

Reply:
Ah I see you're suggesting I compose these two suggestions together, ok can do 👍🏼 Thanks for the suggestions

Reply:
Done in 491425ab90ce7dfdc556249806debafecddab177